p2panda-net/src/engine/engine.rs

```diff
@@ -224,6 +266,11 @@ where
             ToEngineActor::GossipJoined { topic_id } => {
                 self.on_gossip_joined(topic_id).await;
             }
+            ToEngineActor::GossipNeighborDown { peer } => {
```

Comment:
`GossipNeighborDown` indicates that we've lost connection with a direct neighbour in the HyParView cluster. We use that event to initiate a reset of the completed sync state for the peer with whom we lost a connection. This means that we'll be all-clear to initiate sync with that peer when we receive their next topic announcement via the network-wide gossip overlay (ie. whenever they come back online).

Thanks to @sandreae for suggesting this addition!

Reply:
Neighbor down events do not necessarily imply that the peer went offline, so it might mean we initiate sync without any new outcomes, on top we might do that more often as there's potentially multiple gossip overlays. Of course it _can_ sometimes help if that peer went actually offline and _maybe_ came up again in our strong ebt links, but that's a lot of maybes.

Im undecided if this is really what we want?